src/json_parser.cc

```diff
@@ -101,4 +102,51 @@ std::optional<bool> JSONParser::GetTopLevelBoolField(std::string_view field) {
   return value->BooleanValue(isolate);
 }
 
+std::optional<JSONParser::StringDict> JSONParser::GetTopLevelStringDict(
+    std::string_view field) {
+  Isolate* isolate = isolate_.get();
+  v8::HandleScope handle_scope(isolate);
+  Local<Context> context = context_.Get(isolate);
+  Local<Object> content_object = content_.Get(isolate);
+  Local<Value> value;
+  bool has_field;
+  // It's not a real script, so don't print the source line.
+  errors::PrinterTryCatch bootstrapCatch(
+      isolate, errors::PrinterTryCatch::kDontPrintSourceLine);
+  Local<Value> field_local;
+  if (!ToV8Value(context, field, isolate).ToLocal(&field_local)) {
+    return std::nullopt;
+  }
+  if (!content_object->Has(context, field_local).To(&has_field)) {
+    return std::nullopt;
+  }
+  if (!has_field) {
+    return StringDict();
```

Comment:
I just noticed that the logic in SEA config parsing was fine, the bugs was in `GetTopLevelStringDict()` which should've returned empty maps explicitly instead of `{}` which actually gets converts to `std::nullopt` instead.